sia/miner_test.go

```diff
@@ -19,3 +19,12 @@ func mineSingleBlock(t *testing.T, c *Core) {
 		t.Error(err)
 	}
 }
+
+func testMinerDeadlocking(t *testing.T, c *Core) {
+	threads := 1
+	c.MinerInfo()
+	c.UpdateMiner(threads)
+	c.StartMining()
+	c.MinerInfo()
+	c.StopMining()
+}
```

Comment:
It makes sure that all of the calls are actually doing something, and does protect/detect future locking errors. And it actually did catch the race condition above which I hadn't noticed before.

I actually want to implement something like this for every component, just slam all of the functions in a semi-random order and try to trigger a deadlock. It's a good way to add certainty to your mutex code, especially because the race library can only detect a race condition if you trigger it during runtime.